spyder/utils/qthelpers.py

```diff
@@ -751,7 +755,7 @@ def event(self, event):
                 pass
             elif self._has_started:
                 self.sig_open_external_file.emit(fname)
-            elif MAC_APP_NAME not in fname:
+            else:

```

Comment:
You should check that this doesn’t open the exec file in spyder at startup. (Spyder.app/…/spyder would be opened during the startup)

Reply:
I can confirm that no internal files are opened when launching from the bundle or from command line, with or without preexisting open files or projects.